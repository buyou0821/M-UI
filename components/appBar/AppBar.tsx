import React, { forwardRef } from 'react';
import clsx from 'clsx';
import { tuple } from '../_util/type';
import { usePrefixCls } from '../_util/hooks';
import { IconButton, Typography, ToolBar } from './AppBarInner';

const AppBarColors = tuple('primary', 'secondary', 'success', 'warning', 'danger');
export type AppBarColor = (typeof AppBarColors)[number];

interface AppBarProps {
  className?: string;
  children?: React.ReactNode;
  color?: AppBarColor;
  style?: React.CSSProperties;
}

interface AppBarComponent<p> extends React.ForwardRefExoticComponent<p> {
  IconButton: typeof IconButton;
  Typography: typeof Typography;
  ToolBar: typeof ToolBar;
}

const AppBar = forwardRef((props: AppBarProps, ref: React.RefObject<HTMLDivElement>) => {
  const { color = 'default', className, children, ...rest } = props;
  const prefixCls = usePrefixCls('appbar');

  const classes = clsx(prefixCls, className, {
    [`${prefixCls}--${color}`]: color,
  });

  return (
    <div ref={ref} className={classes} {...rest}>
      {children}
    </div>
  );
}) as AppBarComponent<AppBarProps>;

AppBar.IconButton = IconButton;
AppBar.Typography = Typography;
AppBar.ToolBar = ToolBar;

export default AppBar;
