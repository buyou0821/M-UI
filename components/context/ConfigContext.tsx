import React, { createContext, Context } from 'react';

interface ConfigProviderProps {
  prefixCls?: string;
}

interface ConfigConsumerProps {
  prefixCls?: string;
  getPrefixCls?: (suffixCls: string) => string;
}

export const ConfigContext: Context<ConfigConsumerProps> = createContext({
  getPrefixCls: () => {
    const prefixCls = 'tyche';
    // const { prefixCls = 'tyche' } = props;
    // console.log(prefixCls);
    return prefixCls;
  },
});

const ConfigProvider: React.FunctionComponent<ConfigProviderProps> = props => {
  const getPrefixCls = (suffixCls: string) => {
    const { prefixCls = 'tyche' } = props;
    return `${prefixCls}-${suffixCls}`;
  };
  const config: ConfigConsumerProps = {
    getPrefixCls,
  };
  return <ConfigContext.Provider value={config}>{props.children}</ConfigContext.Provider>;
};

export default ConfigProvider;
