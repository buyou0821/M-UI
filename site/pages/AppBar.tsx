import React from 'react';
import { AppBar, Icon, Button } from 'components';

const { IconButton, Typography, ToolBar } = AppBar;

export default () => (
  <AppBar color="primary" className="demo-layout-appBar">
    <IconButton md={0}>
      <Icon style={{ color: '#fff', fontSize: 22 }} type="dehaze" />
    </IconButton>
    <Typography>
      <Icon type="tycheUI" style={{ fontSize: 32, marginRight: 12 }} />
      Tyche UI
    </Typography>
    <ToolBar>
      <Button shape="icon" href="https://github.com/buyou0821/tyche-ui" target="_blank">
        <Icon style={{ color: '#fff', fontSize: 24 }} type="github" />
      </Button>
    </ToolBar>
  </AppBar>
);
