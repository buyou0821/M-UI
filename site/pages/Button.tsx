import React from 'react';
import { Icon, Button, ConfigProvider } from 'components/index';

export default () => {
  return (
    <ConfigProvider>
      <div className="demo-buttons">
        <h3>Contained Buttons</h3>
        <Button>DEFAULT</Button>
        <Button loading color="primary">
          PRIMARY
        </Button>
        <Button color="secondary" disabled>
          SECONDARY
        </Button>
        <Button href="https://github.com/buyou0821/tyche-ui" target="_blank" color="success">
          Tyche UI
        </Button>
        <Button htmlType="submit" color="warning" ripple={false}>
          WARNING
        </Button>
        <Button color="error">ERROR</Button>
      </div>
      <div className="demo-buttons">
        <h3>Text Buttons</h3>
        <Button shape="text">DEFAULT</Button>
        <Button shape="text" color="primary" center>
          <Icon left material="create" />
          PRIMARY
        </Button>
        <Button shape="text" color="secondary">
          SECONDARY
        </Button>
        <Button shape="text" color="success">
          SUCCESS
        </Button>
        <Button shape="text" color="warning">
          WARNING
        </Button>
        <Button shape="text" color="error">
          ERROR
        </Button>
      </div>
      <div className="demo-buttons">
        <h3>Outlined Buttons</h3>
        <Button shape="outlined">DEFAULT</Button>
        <Button shape="outlined" color="primary">
          PRIMARY
        </Button>
        <Button shape="outlined" color="secondary" disabled>
          SECONDARY
        </Button>
        <Button shape="outlined" color="success">
          SUCCESS
        </Button>
        <Button shape="outlined" color="warning">
          WARNING
        </Button>
        <Button shape="outlined" color="error">
          ERROR
        </Button>
      </div>
      <div className="demo-buttons">
        <h3>Circle/Fab Buttons</h3>
        <Button shape="circle">
          <Icon material="face" />
        </Button>
        <Button shape="circle" loading color="primary">
          <Icon material="remove" />
        </Button>
        <Button shape="circle" color="secondary" disabled>
          <Icon material="create" />
        </Button>
        <Button shape="circle" color="success" fab>
          <Icon material="done" />
        </Button>
        <Button shape="circle" color="warning" fab loading>
          <Icon material="priority_high" />
        </Button>
        <Button shape="circle" color="error" fab disabled>
          <Icon material="clear" />
        </Button>
      </div>
      <div className="demo-buttons">
        <h3>Round Buttons</h3>
        <Button shape="round">DEFAULT</Button>
        <Button shape="round" color="primary">
          PRIMARY
        </Button>
        <Button shape="round" color="secondary" disabled>
          SECONDARY
        </Button>
        <Button shape="round" color="success">
          SUCCESS
        </Button>
        <Button shape="round" color="warning">
          WARNING
        </Button>
        <Button shape="round" color="error">
          ERROR
        </Button>
      </div>
      <div className="demo-buttons">
        <h3>Icon Buttons</h3>
        <Button shape="icon">
          <Icon material="face" />
        </Button>
        <Button shape="icon" color="primary">
          <Icon material="remove" />
        </Button>
        <Button shape="icon" color="secondary" disabled>
          <Icon material="create" />
        </Button>
        <Button shape="icon" color="success" size="small">
          <Icon material="done" />
        </Button>
        <Button shape="icon" color="warning">
          <Icon material="priority_high" />
        </Button>
        <Button shape="icon" color="error" size="large">
          <Icon material="clear" />
        </Button>
      </div>
      <div className="demo-buttons">
        <h3>Icon Buttons</h3>
        <Button color="primary">
          <Icon left material="delete" />
          删除
        </Button>
        <Button color="secondary" disabled>
          <Icon left material="create" />
          添加
        </Button>
        <Button color="success">
          成功
          <Icon right material="done" />
        </Button>
        <Button color="warning">
          警告
          <Icon right material="warning" />
        </Button>
        <Button color="error">
          <Icon left material="send" />
          SEND
        </Button>
      </div>
      <div className="demo-buttons">
        <h3>Size</h3>
        <Button color="primary" size="small">
          <Icon left material="delete" />
          删除
        </Button>
        <Button color="secondary">
          <Icon left material="create" />
          添加
        </Button>
        <Button color="success" size="large">
          成功
          <Icon right material="done" />
        </Button>
        <Button color="error" shape="circle" size="small">
          <Icon material="warning" />
        </Button>
        <Button color="warning" shape="circle">
          <Icon material="warning" />
        </Button>
        <Button color="error" shape="circle" size="large">
          <Icon material="warning" />
        </Button>
      </div>
    </ConfigProvider>
  );
};
