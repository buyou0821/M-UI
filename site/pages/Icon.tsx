import React from 'react';
import { Icon } from 'components/index';

export default () => {
  const IconFont = Icon.createFromIconfont({
    scriptUrl: '//at.alicdn.com/t/font_8d5l8fzk5b87iudi.js',
  });
  return (
    <div>
      <Icon type="loading" style={{ color: '#1890ff' }} />
      <Icon material="face" style={{ color: '#f66' }} rotate={90} />
      <Icon material="face" rotate={90} />
      <IconFont type="icon-twitter" spin />
      <IconFont type="icon-tuichu" rotate={90} />
    </div>
  );
};
