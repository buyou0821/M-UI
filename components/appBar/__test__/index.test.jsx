import React from 'react';
import { shallow, render } from 'enzyme';
import { defaultPrefixCls } from '../../context/ConfigContext';
import AppBar from '../index';
import Icon from '../../icon';
import Button from '../../button';

const prefixClx = `${defaultPrefixCls}-appbar`;

describe('AppBar', () => {
  it('should render correctly', () => {
    const { IconButton, Typography, ToolBar } = AppBar;
    const wrapper = render(
      <AppBar color="primary">
        <IconButton>
          <Icon style={{ fontSize: 22 }} type="list" />
        </IconButton>
        <Typography>Title</Typography>
        <ToolBar>
          <Button shape="text" style={{ height: '100%', color: '#fff', marginRight: '-16px' }}>
            Login
          </Button>
        </ToolBar>
      </AppBar>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should support color attribute', () => {
    const wrapper = shallow(<AppBar color="success" />);
    expect(wrapper.find(`.${prefixClx}--success`).length).toEqual(1);
  });
});
