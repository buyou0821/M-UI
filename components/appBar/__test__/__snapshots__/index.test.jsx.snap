// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar should render correctly 1`] = `
<div
  class="ty-appbar ty-appbar--primary"
>
  <div
    class="ty-col ty-appbar__icon"
  >
    <button
      class="ty-ripple ty-btn ty-appbar__icon-btn ty-btn__icon ty-btn__icon--default"
    >
      <span
        class="ty-btn__wrapper"
      >
        <i
          class="ty-icon"
          style="font-size:22px"
        >
          <svg
            aria-hidden="true"
            class=""
            fill="currentColor"
            height="1em"
            width="1em"
          >
            <use
              href="#list"
            />
          </svg>
        </i>
      </span>
      <span
        class="ty-ripple__wrapper"
      />
    </button>
  </div>
  <div
    class="ty-appbar__typography"
  >
    Title
  </div>
  <div
    class="ty-appbar__toolBar"
  >
    <button
      class="ty-ripple ty-btn ty-btn__text ty-btn__text--default"
      style="height:100%;color:#fff;margin-right:-16px"
    >
      <span
        class="ty-btn__wrapper"
      >
        Login
      </span>
      <span
        class="ty-ripple__wrapper"
      />
    </button>
  </div>
</div>
`;
